lib/Sema/ConstraintSystem.cpp

```diff
@@ -2924,8 +2924,10 @@ static bool diagnoseContextualFunctionCallGenericAmbiguity(
   // from all the closure contextual fix/solutions and if there are more than
   // one fixed type diagnose it.
   swift::SmallSetVector<Type, 4> genericParamInferredTypes;
-  for (auto &fix : contextualFixes)
-    genericParamInferredTypes.insert(fix.first->getFixedType(resultTypeVar));
+  for (auto &fix : contextualFixes) {
+    if (fix.first->hasFixedType(resultTypeVar))
```

Comment:
Free type variables are allowed only with `FreeTypeVariableBinding::Allow` set, I think a better fix here might be to actually get an overload from every solution instead and extra their result type because if `hasFixedType` is ever produces true it could mean that overloads were chosen in sub-scope when solutions diverged.

Reply:
> extra their result type

This should say "extract" right?

Reply:
Yes, sorry!